mlflow/models/evaluation/base.py

```diff
@@ -219,6 +229,24 @@ def __init__(self, data, labels, name=None, path=None):
         self.path = path
         self._hash = None
 
+        if isinstance(self.data, np.ndarray):
+            num_features = self.data.shape[1]
+            if feature_names is not None:
+                feature_names = list(feature_names)
+                if num_features != len(feature_names):
+                    raise ValueError('feature name list must be the same length with feature data.')
+                self._feature_names = feature_names
+            else:
+                self._feature_names = [f'f_{i}' for i in range(num_features)]
+        else:
+            pd_column_names = [c for c in self.data.columns if c != self.labels]
+            if feature_names is not None:
+                feature_names = list(feature_names)
+                if pd_column_names != list(feature_names):
```

Comment:
do we want to do a set comparison here to prevent list traversal for large feature sets?